plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java

```diff
@@ -973,9 +979,35 @@ private AWSCredentialsProvider createAwsCredentialsProvider(URI uri, Configurati
                 .orElseGet(DefaultAWSCredentialsProviderChain::getInstance);
 
         if (iamRole != null) {
+            String stsEndpointOverwrite = conf.get(S3_STS_ENDPOINT);
+            String stsRegionOverwrite = conf.get(S3_STS_REGION);
+
+            AWSSecurityTokenServiceClientBuilder stsClientBuilder =
+                    AWSSecurityTokenServiceClientBuilder.standard()
+                    .withCredentials(provider);
+
+            if (!isNullOrEmpty(stsEndpointOverwrite) && !isNullOrEmpty(stsRegionOverwrite)) {
+                stsClientBuilder.withEndpointConfiguration(
+                    new EndpointConfiguration(stsEndpointOverwrite, stsRegionOverwrite));
+            }
+            else if (!isNullOrEmpty(stsRegionOverwrite)) {
```

Comment:
what about the case when `stsEndpointOverwrite` is defined by `stsRegionOverwrite` not. 
We should not silently ignore it. We should either consume `stsEndpointOverwrite` or throw if such configuration is not supported.